contrib/platform/src/com/sun/jna/platform/win32/COM/Wbemcli.java

```diff
@@ -194,6 +212,43 @@ public IWbemQualifierSet GetPropertyQualifierSet(String strProperty) {
             IWbemQualifierSet qualifier  = new IWbemQualifierSet(ppQualSet.getValue());
             return qualifier;
         }
+
+        public HRESULT Put(String wszName, int lFlags, Variant.VARIANT pVal, int Type) {
+            return Put(wszName == null ? null : new WString(wszName), lFlags, pVal, Type);
+        }
+
+        public HRESULT Put(WString wszName, int lFlags, Variant.VARIANT pVal, int Type) {
+            // 6th method in IWbemClassObjectVtbl
+            return (HRESULT) _invokeNativeObject(5,
+                new Object[]{ getPointer(), wszName, lFlags, pVal, Type}, HRESULT.class);
+        }
+
+        public void Put(String wszName, String pValue) {
+            Variant.VARIANT aVariant = new Variant.VARIANT();
+            BSTR strValue = OleAuto.INSTANCE.SysAllocString(pValue);
+            try {
+                aVariant.setValue(Variant.VT_BSTR, strValue);
+                HRESULT res = Put(wszName, 0, aVariant, 0);
+                COMUtils.checkRC(res);
+            }
+            finally {
+                OleAuto.INSTANCE.SysFreeString(strValue);
```

Comment:
This should not be necessary. From documentation (https://learn.microsoft.com/en-us/windows/win32/api/oleauto/nf-oleauto-variantclear):

> The function clears a VARIANTARG by setting the vt field to VT_EMPTY. The current contents of the VARIANTARG are released first. **If the vtfield is VT_BSTR, the string is freed**. If the vtfield is VT_DISPATCH, the object is released. If the vt field has the VT_ARRAY bit set, the array is freed.

I emphasized the critical part.

Reply:
I removed the call to SysFreeString and left only the VariantClear call here